opencog/cogserver/server/AgentRunnerThread.cc

```diff
@@ -0,0 +1,167 @@
+/*
+ * opencog/server/AgentRunnerThread.cc
+ *
+ * This program is free software; you can redistribute it and/or modify
+ * it under the terms of the GNU Affero General Public License v3 as
+ * published by the Free Software Foundation and including the exceptions
+ * at http://opencog.org/wiki/Licenses
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU Affero General Public License
+ * along with this program; if not, write to:
+ * Free Software Foundation, Inc.,
+ * 51 Franklin Street, Fifth Floor, Boston, MA 02110-1301 USA.
+ */
+
+#include <opencog/cogserver/server/AgentRunnerThread.h>
+#include <opencog/util/Logger.h>
+
+using namespace std;
+
+
+namespace opencog
+{
+
+AgentRunnerThread::AgentRunnerThread(const std::string &name) :
+        AgentRunnerBase(name), runAgents(false), agentsModified(false),
+        clearAll(false)
+{
+}
+
+AgentRunnerThread::~AgentRunnerThread()
+{
+    removeAllAgents();
+    joinRunThread();
+}
+
+void AgentRunnerThread::start()
+{
+    {
+        // locking is required, to prevent dead-lock in condition_variable wait()
+        lock_guard<mutex> lock(runAgentsMutex);
+        runAgents.store(true, memory_order_relaxed);
+    }
+    runAgentsCond.notify_one();
+}
+
+void AgentRunnerThread::stop()
+{
+    runAgents.store(false, memory_order_relaxed);
```

Comment:
It is an atomic_bool, and I wanted it to be atomic here but I don't want to enforce any memory order here. I think nobody would care if the thread is stopped slightly later. `runAgents = false;` is store() with strict memory order.